### PR TITLE
Documentation: part 4 (scopes/media-remap/install rework, and tutorial rewrite)

### PR DESCRIPTION
**Pull request by ubitux**
Depends on:
- https://github.com/NopeFoundry/nope.gl/pull/135
- https://github.com/NopeFoundry/nope.gl/pull/138
- https://github.com/NopeFoundry/nope.gl/pull/137

### CODE REVIEW

File: doc/_ext/nope/scopes/histogram.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from pynopegl_utils.misc import SceneCfg, scene
+
+import pynopegl as ngl
+
+
+@scene()
+def histogram(cfg: SceneCfg):
+    jpegs = (m for m in cfg.medias if m.filename.endswith(".jpg"))
```
Comment (mbouron) [inline doc/_ext/nope/scopes/histogram.py:8]: images + (jpg or png)

File: doc/_ext/nope/scopes/waveform.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from pynopegl_utils.misc import SceneCfg, scene
+
+import pynopegl as ngl
+
+
+@scene()
+def waveform(cfg: SceneCfg):
+    jpegs = (m for m in cfg.medias if m.filename.endswith(".jpg"))
```
Comment (mbouron) [inline doc/_ext/nope/scopes/waveform.py:8]: Same ?

File: doc/usr/tuto/start.md [hunk 3]
```diff
@@ -1,431 +1,112 @@
...
+Hello World!
 ```
 
-![my reddish demo](img/ngl-control-reddish-scene.png)
+Loading that script in the viewer should provide the same rendering observed
+here.
 
-[book-of-shaders]: http://thebookofshaders.com/
-[expl-shaders]: /usr/expl/shaders.md
+In this first example all we did was to return a single node (the [Text] node),
```
Comment (mbouron) [inline doc/usr/tuto/start.md:31]: In this first example, all we did was to declare declare a function that return a single node (...). The `@scene()` decorator is used to make it discoverable by the viewer.

File: doc/usr/tuto/start.md [hunk 4]
```diff
@@ -1,431 +1,112 @@
...
+in a `@scene()` decorated function for the viewer to find it. The function
+receives a configuration of type `SceneCfg` which gives us access to various
+metadata on the scene (for both reading and sometimes writing).
```
Comment (mbouron) [inline doc/usr/tuto/start.md:34]: - of the scene ?
- maybe list a few metata ?

File: doc/usr/tuto/start.md [hunk 5]
```diff
@@ -1,431 +1,112 @@
...
+Animating the background colors
 ```
 
-![color widget](img/ngl-control-color-widget.png)
-
-All the other widgets are documented in the [Controller widgets
-documentation][controller-widgets].
-
-[controller-widgets]: /usr/ref/pynopegl-utils.md#controller-widgets
+`color0_animkf` and `color1_animkf` define time key frames: they associate
```
Comment (mbouron) [inline doc/usr/tuto/start.md:68]: `define time key frames:` `define time key frames. Time key frames ...`

File: doc/usr/tuto/start.md [hunk 6]
```diff
@@ -1,431 +1,112 @@
...
+a time with a value, and the engine will interpolate the intermediate values
```
Comment (mbouron) [inline doc/usr/tuto/start.md:69]: `, and the` -> `for which the`

**Review by mbouron — APPROVED** [review]
Other than my comments, LGTM.

File: doc/_ext/nope/scopes/histogram.py (hunk 1 above)
Comment (ubitux) [reply doc/_ext/nope/scopes/histogram.py:8]: Same as previous PR, I'm looking for a rich picture (to have significant histogram and waveforms)

File: doc/_ext/nope/scopes/histogram.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    img = next(jpegs)
```
[the remark this answers is not in the document]
Comment (ubitux) [reply doc/_ext/nope/scopes/histogram.py]: Renamed

File: doc/_ext/nope/scopes/waveform.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    img = next(jpegs)
```
[the remark this answers is not in the document]
Comment (ubitux) [reply doc/_ext/nope/scopes/waveform.py]: Renamed

File: doc/usr/tuto/start.md (hunk 3 above)
Comment (ubitux) [reply doc/usr/tuto/start.md]: Changed, thanks. (removed the duped "declare" and `+s` on "return")

File: doc/usr/tuto/start.md (hunk 4 above)
Comment (ubitux) [reply doc/usr/tuto/start.md]: Done

File: doc/usr/tuto/start.md [hunk 9]
```diff
@@ -1,431 +1,112 @@
...
+play with that value (somewhere between 0 and 1) to see how it impacts the scene.
 
-![uniform color demo](img/graph-uniform-color-demo.png)
+## Animating values
 
-### Scene widgets
+Many nodes that accept constants also allow nodes as input, which means they can
+be animated. Looking at the [RenderGradient] parameters, we see many parameters
+with the `node` flag, notably the `color1` and `color2` we're currently using,
+which means we could animate them.
```
[the remark this answers is not in the document]
Comment (ubitux) [reply doc/usr/tuto/start.md:63]: That's the current section though 🤔

File: doc/usr/tuto/start.md (hunk 5 above)
Comment (ubitux) [reply doc/usr/tuto/start.md]: Replaced with:

> `color0_animkf` and `color1_animkf` define time key frames. Each of these time key frames associate a time with a value ...`

File: doc/usr/tuto/start.md (hunk 6 above)
Comment (ubitux) [reply doc/usr/tuto/start.md]: Done

File: doc/usr/tuto/start.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (mbouron) [reply doc/usr/tuto/start.md:63]: `which means we could animate them.` <- I find this part redundant with the previous part `which means they can be animated`

[the remark this answers is not in the document]
Comment (ubitux) [reply doc/usr/tuto/start.md]: Ooh you're right, completely missed that. How about the following?

```markdown
In many cases, constant node parameters also allow nodes. We can see that it's
the case with [RenderGradient] parameters (look for the `node` flag), notably
the `color1` and `color2` we're currently using in our scene. Accepting nodes
means we could animate these parameters:
```

Branch has been updated.

[the remark this answers is not in the document]
Comment (mbouron) [reply doc/usr/tuto/start.md]: `In many cases, constant node parameters also allow nodes.` -> `In many cases, node parameters that accept constant values also accept nodes.`

[the remark this answers is not in the document]
Comment (mbouron) [reply doc/usr/tuto/start.md]: `, notably` `and notably`

[the remark this answers is not in the document]
Comment (ubitux) [reply doc/usr/tuto/start.md]: Thanks! changed.

**Pull request merged**